pkg/storage/sds/edvformatprovider/edvformatprovider.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package edvformatprovider
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+	"github.com/hyperledger/aries-framework-go/pkg/storage/sds"
+	"github.com/hyperledger/aries-framework-go/pkg/storage/sds/edvformatprovider/edvdocumentprocessor"
```

Comment:
edvformatprovider depends on a sub package (edvdocumentprocessor), is the document processor only going to be used by the format provider?

Reply:
Yes. The EDVFormatProvider will use it to do the encryption before storing in the underlying provider, and it'll also use it for decrypting the data from the underlying provider. The idea is that the existing providers work without making any changes. In another PR I'll be pushing in the near-future, I'll be adding the EDV rest API as a provider as well that can be used as an underlying provider.

Reply:
(per our offline conversation, I made the EDVFormatProvider and EDVDocumentProcessor packages siblings in terms of their folder hierarchy.)